### DISCUSSION TITLE
RustDesk Portable should be really portable (with example of possible simple solution in code)

### DISCUSSION THREAD

**Discussion by pedjas** [rustdesk/rustdesk#2127] [General]
RustDesk Portable is not actually portable. It saves configuration in $APPDATA%/Rust directory regardless where exe is located.

Real portable applications keep their configuration and data with exe, so it can simply copied and run anywhere. RustDesk Portable should do the same. That should also resolve issue of having two or more RusteDesk portable executables on the same PC share the same configuration which causes they interfere each other and cannot work simultaneously.

**Comment by pedjas** [comment]
I am not familiar to Rust so it took me a while until I found out how it loads configuration. From what I see, it could be simple change:

in config.rs, class Config, method path, line 403 is:

`            if let Some(project) = directories_next::ProjectDirs::from("", &org, &*APP_NAME.read().unwrap()) {`

That finds out system roaming directory, then adds application name to it to form path where to store and load config from. That is issue as all RustDesk Clients on computer use the same configuration and interfere each other.

That could be replaced with call to std::env::current_exe() to get path of the executable, I guess in combination with executable_path::executable_path(), to get directory where RustDesk Client is run from. That would provide that each executable has its own independent configuration.

http://web.mit.edu/rust-lang_v1.25/arch/amd64_ubuntu1404/share/doc/rust/html/std/env/fn.current_exe.html
https://docs.rs/executable-path/latest/executable_path/fn.executable_path.html

**Comment by zhpjy** [comment]
Need this feature too
